### PR TITLE
#54 - Push image manifest endpoint added

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #54 
Added push image manifest endpoint

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#82](https://github.com/artipie/docker-adapter/issues/82) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/681fedf5-854f-484b-8369-d584bb3c2f13); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/docker/http/PushImageManifest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import com.artipie.http.rs.RsStatus;
+import com.artipie.http.rs.RsWithStatus;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.regex.Pattern;
+import org.reactivestreams.Publisher;
+
+/**
+ * Slice for push image manifest endpoint.
+ * See <a href="https://docs.docker.com/registry/spec/api/#pushing-an-image">Pushing An Image</a>.
```
Comment (Vatavuk) [inline src/main/java/com/artipie/docker/http/PushImageManifest.java:37]: @olegmoz change url to `https://docs.docker.com/registry/spec/api/#pushing-an-image-manifest`

Comment (olegmoz) [reply src/main/java/com/artipie/docker/http/PushImageManifest.java]: @Vatavuk fixed it, thanks

**Comment by 0crat** [comment]
Job was finished in 10 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**